### DISCUSSION TITLE
Categorization

### DISCUSSION THREAD

**Discussion by garyrob** [ananthakumaran/paisa#25] [Ideas]
You said on Hacker News: "It has a very crude tf-idf based categorization. I do have plans to improve it." 

I responded there as follows, but I'm posting here too in case it's helpful"
You could use the same algorithm that was used for spam filtering in my Linux Journal article from back in 2003[1]. I have thought about making a plugin to do it with MoneyDance but haven't gotten around to it. But I think it would be quite easy for you to integrate into Paisa if you are looking to do that sort of thing. I have actually made some improvements to the algorithm since then; let me know if you're interested...

[1] https://www.linuxjournal.com/article/6467"

**Comment by ananthakumaran** [comment]
I will check the article later, just skimmed it, but it seems to talk about binary classification. In Paisa case, it has to be multi-class classification.

**Comment by garyrob** [comment]
Yes, but it could be used for multi-class as well. Let me know if you want thoughts about it...
